src/game/SpellEffects.cpp

```diff
@@ -7147,10 +7155,20 @@ void Spell::EffectSummonDeadPet(SpellEffectIndex /*eff_idx*/)
 {
     if (m_caster->GetTypeId() != TYPEID_PLAYER)
         return;
+
+    bool wasDespawned = false;
     Player* _player = (Player*)m_caster;
     Pet* pet = _player->GetPet();
     if (!pet)
-        return;
+    {
+        pet = new Pet;
+        if (!pet->LoadPetFromDB(_player, 0, 0, false, true))
```

Comment:
If we can't load pet doesn't that mean there is no pet, not that it has been despawned? - also, it should be possible to resurrect your pet when it's not around, while it shouldn't be possible to call it close at all if it's dead.

not quite sure what exactly you're checking for here?